rfcs/react-components/styles-handbook.md

```diff
@@ -0,0 +1,922 @@
+
+- [Introduction](#introduction)
+- [Core concepts](#core-concepts)
+  - [Atomic CSS](#atomic-css)
+  - [`makeStyles`](#makestyles)
+    - [Limitations](#limitations)
+    - [Performance caveat](#performance-caveat)
+  - [`mergeClasses()`](#mergeclasses)
+    - [Order of arguments determines results](#order-of-arguments-determines-results)
+    - [⚠️ Only combine classes with `mergeClasses`](#%EF%B8%8F-only-combine-classes-with-mergeclasses)
+  - [`makeResetStyles`](#makeresetstyles)
+    - [Hybrid approach (Using `makeStyles` and `makeResetStyles` together)](#hybrid-approach-using-makestyles-and-makeresetstyles-together)
+  - [RTL styles](#rtl-styles)
+    - [CSS variables caveat](#css-variables-caveat)
+  - [Understanding selector complexity](#understanding-selector-complexity)
+- [Best practices](#best-practices)
+  - [Writing styles](#writing-styles)
+    - [Use `tokens` over direct colors](#use-tokens-over-direct-colors)
+    - [Avoid rule duplication](#avoid-rule-duplication)
+    - [Avoid `!important`](#avoid-important)
+    - [Use structured styles](#use-structured-styles)
+  - [Performance](#performance)
+    - [Use `mergeClasses` once for an element](#use-mergeclasses-once-for-an-element)
+    - [Avoid unnecessary RTL transforms by using `@noflip`](#avoid-unnecessary-rtl-transforms-by-using-noflip)
+  - [Nested selectors](#nested-selectors)
+    - [Use nested selectors with pseudo classes](#use-nested-selectors-with-pseudo-classes)
+    - [Apply classes directly to elements](#apply-classes-directly-to-elements)
+    - [Avoid complicated selectors](#avoid-complicated-selectors)
+    - [Avoid input pseudo classes](#avoid-input-pseudo-classes)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Introduction
+
+Griffel is a hybrid CSS-in-JS that features runtime option like any other CSS-in-JS and [Ahead-of-time compilation][griffel-aot] with [CSS extraction][griffel-css-extraction] to reduce runtime footprint and improve performance.
+
+# Core concepts
+
+## Atomic CSS
+
+Griffel uses Atomic CSS to generate classes. In Atomic CSS every property-value is written as a single CSS rule.
+
+```css
+/* Monolithic classes */
+/* Can be applied only to a specific button */
+.button {
+  display: flex;
+  align-items: center;
+}
+
+/* Atomic CSS */
+/* Can be applied to any element that needs these rules */
+.display-flex {
+  display: flex;
+}
+.align-items-center {
+  align-items: center;
+}
+```
+
+[Learn more][griffel-atomic-css] about Atomic CSS.
+
+> 💡 **Note:** All examples in this document use `@griffel/react` package. However, if you're a Fluent UI consumer please use `@fluentui/react-components` in imports.
+
+## `makeStyles`
+
+`makeStyles` is used to define style permutations in components and is used for style overrides. Returns a [React hook][react-hook] that should be called inside a component:
+
+```js
+import { makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
```

Comment:
Might be nice to show the usage for `useClasses` in this example.

```tsx
const classes = useClasses();

<button className={classes.button} //...
```

This usage is shown in the `mergeClasses` section but it seems to be this is part of the `makeStyles` API.

Reply:
Great idea, updated in d9ba1ba.